lib/Util/Util.php

```diff
@@ -222,15 +257,40 @@ private static function sha256hash($payload)
      */
     private static function normalizeJson($data)
     {
+        $jsonOriginal = json_decode($data, false);
         $json = json_decode($data, true);
         if ($json == null) {
             return $data;
         }
-
         self::jsonSort($json);
+        self::fixEmptyObjectReplacements($jsonOriginal, $json, '');
         return json_encode($json);
     }
 
+    private static function fixEmptyObjectReplacements($originalJson, &$object, $path)
```

Comment:
Could you help me understand what you are trying to do with this block? This looks a bit hacky.

Reply:
Yes, that's a little bit hacky, but without it normalizing JSON was broken. When deserializing for sort json by key names and serializing it back we were having empty arrays instead of empty object. There is a way to force serializing those nodes as empty object, but that serializes empty arrays as objects as well. So what it does - simply search for all empty objects in original JSON and makes according changes to result JSON.
![image](https://user-images.githubusercontent.com/3582353/49291916-6de7d280-f4b4-11e8-93c1-c0eea097f2c1.png)

Reply:
You can use the `JSON_FORCE_OBJECT` option to decode and stop calling ksort when you get an empty object(instead of an array). There has to be a way to implement this correctly without a hack.

Reply:
`JSON_FORCE_OBJECT` is what I meant in my previous comment. It forces empty arrays to be converted to empty objects as well, which is also an issue.

Reply:
you are right but there has to be a way to implement this correctly without a hack.

Reply:
Me and Vlad tried few libraries which do nothing more but wrapper around `json_decode` which doesn't help. So only solution for sorting and normalizing json correctly we came up with is this one.